plugins/snowflake/credentials.go

```diff
@@ -0,0 +1,108 @@
+package snowflake
+
+import (
+	"context"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func Credentials() schema.CredentialType {
+	return schema.CredentialType{
+		Name:    credname.Credentials,
+		DocsURL: sdk.URL("https://docs.snowflake.com/en/user-guide/snowsql.html"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Account,
+				MarkdownDescription: "Snowflake account name.",
+				Optional:            false,
+				Secret:              false,
+				Composition: &schema.ValueComposition{
+					Charset: schema.Charset{
+						Lowercase: true,
+						Uppercase: true,
+						Digits:    true,
+						Symbols:   true,
+					},
+				},
+			},
+			{
+				Name:                fieldname.Username,
+				MarkdownDescription: "Snowflake username.",
+				Optional:            false,
+				Secret:              false,
+				Composition: &schema.ValueComposition{
+					Charset: schema.Charset{
+						Lowercase: true,
+						Uppercase: true,
+						Digits:    true,
+						Symbols:   true,
+					},
+				},
+			},
+			{
+				Name:                fieldname.Password,
+				MarkdownDescription: "Password used to authenticate to Snowflake account.",
+				Optional:            false,
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Charset: schema.Charset{
+						Lowercase: true,
+						Uppercase: true,
+						Digits:    true,
+						Symbols:   true,
+					},
+				},
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
+		Importer: importer.TryAll(
+			importer.TryEnvVarPair(defaultEnvVarMapping),
+			TrySnowflakeConfigFile(),
+		)}
+}
+
+var defaultEnvVarMapping = map[string]sdk.FieldName{
+	"SNOWSQL_ACCOUNT": fieldname.Account,
+	"SNOWSQL_USER":    fieldname.Username,
+	"SNOWSQL_PWD":     fieldname.Password,
+}
+
+func TrySnowflakeConfigFile() sdk.Importer {
+	return importer.TryFile("~/.snowsql/config", func(ctx context.Context, contents importer.FileContents, in sdk.ImportInput, out *sdk.ImportAttempt) {
+		credentialsFile, err := contents.ToINI()
+		if err != nil {
+			out.AddError(err)
+			return
+		}
+
+		// The generated ~/.snowsql/config file, by default, includes an uncommented [connections.example] credential section below the actual
+		// [connections] section. We save the accountname, username, and password listed under [connections] then stop parsing the file
```

Comment:
Could you add a test case (with fixture) for this scenario (connections.example) as well?

Reply:
Just added two more tests! Please let me know if anything seems off.

Reply:
This looks great!